[14.0][NEW] pos_cash_control

ygcarvalh opened the pull request:
This is a process of backporting functionality that exists in newer versions of Odoo (16).

It is already possible to perform this type of action in Odoo 14, but with the interactions being performed entirely in the backend. This functionality allows the Bleeding/Supply process (Cash In/Cash Out) to be performed entirely by the front-end.

In addition, the cashier closing control process will also be implemented - as is done in Odoo 16 - and which is carried out entirely through the front-end.

mileo commented:
@ygcarvalh could you please change the author of the first commit as said in the guideline:

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#backporting-odoo-modules

Commit:
[IMP] pos_cash_control: backport of cash in/out functionality

ygcarvalh commented:
> @ygcarvalh could you please change the author of the first commit as said in the guideline:
> 
> https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#backporting-odoo-modules

Can you check if it's correct now, please?

ygcarvalh committed:
[IMP] pos_cash_control: Addition of the model to manage bills and coins

ygcarvalh committed:
[IMP] pos_cash_control: Relation between pos.config and pos.bill models

A Boolean field required to define the maximum difference value allowed
during cash closing was also added.

ygcarvalh committed:
[IMP] pos_cash_control: pop ups required for cash closing via front

ygcarvalh committed:
[ADD] pos_cash_control: Functionality for closing the POS via front-end

ygcarvalh committed:
[IMP] pos_cash_control: New POS session closing flow

ygcarvalh committed:
[FIX] pos_cash_control: Update flow for case tests

ygcarvalh committed:
[I18N] pos_cash_control: Portuguese translation

ygcarvalh committed:
[FIX] pos_cash_control: Missing params for other payments

ygcarvalh committed:
[FIX] pos_cash_control: Backport missing parts
